dotCMS/src/main/java/com/dotcms/ai/client/AIProxyClient.java

```diff
@@ -0,0 +1,113 @@
+package com.dotcms.ai.client;
+
+import com.dotcms.ai.client.openai.OpenAIClient;
+import com.dotcms.ai.client.openai.OpenAIResponseEvaluator;
+import com.dotcms.ai.domain.AIProvider;
+import com.dotcms.ai.domain.AIRequest;
+import com.dotcms.ai.domain.AIResponse;
+import io.vavr.Lazy;
+
+import java.io.OutputStream;
+import java.io.Serializable;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * A proxy client for managing and interacting with multiple AI service providers.
+ *
+ * <p>
+ * This class provides a mechanism to send requests to various AI service providers through proxied clients,
+ * applying different strategies for handling the requests and responses. It supports adding new clients and
+ * switching between different AI providers.
+ * </p>
+ *
+ * <p>
+ * The class allows for flexible handling of AI service interactions by maintaining a map of proxied clients
+ * and providing methods to send requests to the current or specified provider.
+ * </p>
+ *
+ * @author vico
+ */
+public class AIProxyClient {
+
+    private static final Lazy<AIProxyClient> INSTANCE = Lazy.of(AIProxyClient::new);
+
+    private final ConcurrentMap<AIProvider, AIProxiedClient> proxiedClients;
```

Comment:
in order to be extendable, we need to use an String instead of an enum, seems the enum wont be easy to change on runtime by osgi
However you can use AIProvider.name() as a key

Reply:
OK. Changing it to use `String` instead of `AIProvider` enum.